src/bunit.core/TestServiceProvider.cs

```diff
@@ -55,20 +55,48 @@ private TestServiceProvider(IServiceCollection initialServiceCollection, bool in
 				serviceProvider = serviceCollection.BuildServiceProvider();
 		}
 
+		/// <summary>
+		/// Add a fall back service provider that provides services when the default returns null
+		/// </summary>
+		/// <param name="serviceProvider">The fallback service provider</param>
+		public void AddFallbackServiceProvider(IServiceProvider serviceProvider)
+		{
+			fallbackServiceProvider = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
+		}
+
 		/// <summary>
 		/// Get service of type T from the test provider.
 		/// </summary>
 		/// <typeparam name="TService">The type of service object to get.</typeparam>
 		/// <returns>A service object of type T or null if there is no such service.</returns>
-		public TService GetService<TService>() => (TService)GetService(typeof(TService));
+		public TService? GetService<TService>() => (TService?)GetService(typeof(TService))!;
 
+#if NETSTANDARD2_1
 		/// <inheritdoc/>
+		[return: MaybeNull]
 		public object GetService(Type serviceType)
+		{
+			return GetServiceInternal(serviceType);
+		}
+#elif NET5_0
+		/// <inheritdoc/>
+		public object? GetService(Type serviceType)
+		{
+			return GetServiceInternal(serviceType);
+		}
+#endif
```

Comment:
```suggestion
		[return: MaybeNull]
		public object GetService(Type serviceType)
			=> GetServiceInternal(serviceType);
#else
		/// <inheritdoc/>
		public object? GetService(Type serviceType)
			=> return GetServiceInternal(serviceType);
#endif
```

Using `#else` here makes it less likely to break in the future when upgrading to next version of .net